src/UpgradeRule/PHP/Visitor/ConstantWarningsVisitor.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace SilverStripe\Upgrader\UpgradeRule\PHP\Visitor;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ConstFetch;
+use SilverStripe\Upgrader\Util\ApiChangeWarningSpec;
+
+/**
+ * Relies on {@link SymbolContextVisitor} to annotate the
+ * 'symbolContext' attribute of nodes.
+ *
+ * @package SilverStripe\Upgrader\UpgradeRule
+ */
+class ConstantWarningsVisitor extends WarningsVisitor
+{
+
+    public function enterNode(Node $node)
+    {
+        parent::enterNode($node);
+
+        if ($node instanceof ConstFetch) {
+            foreach ($this->specs as $spec) {
+                if (!$this->matchesSpec($node, $spec)) {
+                    continue;
+                }
+
+                $this->addWarning($node, $spec);
+            }
+        }
+    }
+
+    /**
+     * @param Node $node
+     * @param ApiChangeWarningSpec $spec
+     * @return bool
+     */
+    protected function matchesSpec(Node $node, ApiChangeWarningSpec $spec)
+    {
+        return ($spec->getSymbol()=== $node->name->parts[0]);
```

Comment:
This compares the symbol to the top level namespace. E.g. `SilverStripe`. Is this intentional?

Docs on `Name::getFirst()`

```php
    /**
     * Gets the first part of the name, i.e. everything before the first namespace separator.
     *
     * @return string First part of the name
     */
    public function getFirst() {
        return $this->parts[0];
    }
```

Reply:
The result of this is that this WILL work for global constants, but NOT for class constants. E.g. `BASE_URL`, but not `SilverStripe\Versioned\Versioned::DRAFT` (should that ever be removed, hypothetically).

Reply:
Fix is to cast `(string)$node->name` and compare that way. :)

Reply:
I don't see how a global constant (not a class constant) can have more than one part in terms of the PHP language, so I'm just using the first part here - that was intentional. Added class constant checks to distinguish that case, was silently ignored before because it's a different node class. Fixed the string casting as you suggested.